ui-frontend/packages/catalog-ui-search/src/main/webapp/component/location-old/location-old.js

```diff
@@ -270,6 +290,158 @@ module.exports = Backbone.AssociatedModel.extend({
     }
   },
 
+  convertLatLonLinePolyToUsng(points) {
+    return points
+      ? points.map((point) => {
+          // A little bit unintuitive, but lat/lon is swapped here
+          return converter.LLtoMGRSUPS(point[1], point[0], usngPrecision)
+        })
+      : undefined
+  },
+
+  convertLatLonLinePolyToDms(points) {
+    return points
+      ? points.map((point) => {
+          const lat = dmsUtils.ddToDmsCoordinateLat(point[1])
+          const lon = dmsUtils.ddToDmsCoordinateLon(point[0])
+          return {
+            lat: lat.coordinate,
+            lon: lon.coordinate,
+            latDirection: lat.direction,
+            lonDirection: lon.direction,
+          }
+        })
+      : undefined
+  },
+
+  convertLatLonLinePolyToUtm(points) {
+    return points
+      ? points.map((point) => {
+          let llPoint = this.LLtoUtmUps(point[1], point[0])
+          return {
+            ...llPoint,
+            hemisphere:
+              llPoint.hemisphere.toUpperCase() === 'NORTHERN'
+                ? 'Northern'
+                : 'Southern',
+          }
+        })
+      : undefined
+  },
+
+  setUsngDmsUtmWithLineOrPoly(model) {
+    let key = this.get('mode')
+    if (key === 'poly') key = 'polygon'
+    if (key && (key === 'line' || key === 'polygon')) {
```

Comment:
✏️ I see that you're updating the key, so you can using it in the model.set (in the other methods), but you might factor out these bits.